### PR TITLE
fix(behavior_path_planner): add shiftedpathcheck() in avoid_module 

### PR DESCRIPTION
**Pull request by shulanbushangshu**
## Description
Solve the problem that the path planned by the avoidance module of behavior_path_planner exceeds the driving area due to the separation of lanes at intersections and the discontinuity of adjacent lanes.

## Related links
https://github.com/autowarefoundation/autoware.universe/issues/1793

## Tests performedpian
(1)Single obstacle

oneobstacle_shiftcheck
(2)Multiple obstacles
twoobstacles_shiftcheck

(3)Dynamic test

https://user-images.githubusercontent.com/102840938/192966777-1eb89193-1a17-4ae0-9168-372b15b984e5.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@satoshi-ota You are trying to solve this kind of issue. Could you review please.

**Comment by mitsudome-r** [comment]
@satoshi-ota Will you be able to review this PR?

**Comment by satoshi-ota** [comment]
> @satoshi-ota Will you be able to review this PR?

Yes, I'll review this PR in this week.

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 1]
```diff
@@ -2663,4 +2674,121 @@ void AvoidanceModule::updateAvoidanceDebugData(
   }
 }
 
+bool AvoidanceModule::shiftedpathcheck(PathShifter & path_shifter, ShiftedPath shifted_path) const
+{
+  const auto & route_handler = planner_data_->route_handler;
+  PathWithLaneId path_refence = shifted_path.path;  //
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:2680]: remove `// ` ?

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 2]
```diff
@@ -2663,4 +2674,121 @@ void AvoidanceModule::updateAvoidanceDebugData(
...
+  auto shift_length = shifted_path.shift_length;
+  lanelet::ConstLanelets current_lanelets;
+  lanelet::ConstLanelets check_lanelets;
+  current_lanelets = avoidance_data_.current_lanelets;
+  static bool is_outside_prv = false;
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:2685]: what does `prv` mean ? Could you add more comments for readability ? :pray:

**Comment by mitsudome-r** [comment]
@shulanbushangshu Are you still working on this PR? If so, could you resolve the conflict?

**Comment by shulanbushangshu** [comment]
@mitsudome-r 
I will test this issue in the newest code. If this issue occurs in the newest code, I will work on this PR and discuss with the community.

**Comment by shulanbushangshu** [comment]
Yes,I update the code to the newest code.It is fixed in the obstacle_avoidance_planner by another way:
Screenshot from 2023-04-13 10-50-40

**Pull request closed without merging**